src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -653,24 +653,15 @@ object ErgoNodeViewHolder {
     val chainUpdateDelay = timeProvider.time() - lastUpdate
     val acceptableChainUpdateDelay = settings.nodeSettings.acceptableChainUpdateDelay
     def chainUpdateDelayed = chainUpdateDelay > acceptableChainUpdateDelay.toMillis
-    def blockUpdateDelayed =
-      history.bestFullBlockOpt
-        .map(b => timeProvider.time() - b.header.timestamp)
-        .exists(blockUpdateDelay => blockUpdateDelay > acceptableChainUpdateDelay.toMillis)
-
     def chainSynced =
       history.bestFullBlockOpt.map(_.id) == history.bestHeaderOpt.map(_.id)
 
-    if (chainUpdateDelayed || blockUpdateDelayed) {
+    if (chainUpdateDelayed) {
       val bestFullBlockOpt =
         history.bestFullBlockOpt
           .filter(_.id != lastMod.id)
           .fold("")(fb => s"\n best full block: $fb")
-      val repairNeeded = if(blockUpdateDelayed) {
-        ErgoHistory.repairIfNeeded(history)
-      } else {
-        false
-      }
+      val repairNeeded = ErgoHistory.repairIfNeeded(history)
```

Comment:
RepairIfNeeded can be called unconditionally as there is a strict condition inside for reparing.